src/include/migraphx/op/allocate.hpp

```diff
@@ -62,26 +62,38 @@ struct allocate
 
     shape compute_shape(const std::vector<shape>& inputs) const
     {
-        if(s != shape())
+        if(s)
```

Comment:
This op. only allows 0 or 1 inputs, correct?  Check that here.

Reply:
Already checked later on with the check_shapes calls